tangelo/linq/target/target_qiskit.py

```diff
@@ -92,11 +96,34 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
             job_sim = self.qiskit.execute(translated_circuit, backend, noise_model=qiskit_noise_model,
                                           shots=self.n_shots, basis_gates=None, optimization_level=opt_level)
             sim_results = job_sim.result()
-            frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
+            self.all_frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
 
-            self.all_frequencies = frequencies.copy()
+            frequencies = self.all_frequencies
```

Comment:
Attention: here `frequencies` and `self.all_frequencies` are now pointing to the same location in memory. Changing one will change the other. In the previous version of the code, for some reason, one was pointing to a copy and thus changing one would not change the other in the subsequent lines.

Are we sure this is the desired behaviour ?